### PR TITLE
Bb/244/upgrade pydantic

### PR DESCRIPTION
**Pull request by iambroadband**
1. Address long-standing issues with our `scryfall_id` alias implementation.
2. Upgrade to support Pydantic 2.6

### CODE REVIEW

File: src/scooze/models/card.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from datetime import date
 from typing import Any
 
-from pydantic import ConfigDict, Field, field_serializer, field_validator
+from pydantic import AliasChoices, ConfigDict, Field, field_serializer, field_validator
```
Comment (iambroadband) [inline src/scooze/models/card.py:4]: This was the biggest part of the fix.

File: src/scooze/models/card.py [hunk 2]
```diff
@@ -174,8 +174,7 @@ class CardModelData(ScoozeBaseModel):
     scryfall_id: str | None = Field(
         default=None,
         description="Scryfall's unique ID for this card.",
-        validation_alias="id",
-        alias_priority=1,
+        validation_alias=AliasChoices("scryfall_id", "scryfallId", "id"),
```
Comment (iambroadband) [inline src/scooze/models/card.py:177]: This was the biggest part of the fix.

File: tests/routers/conftest.py [hunk 3]
```diff
@@ -19,24 +19,14 @@ def mock_scooze_client():
 def mock_cards_collection(mock_scooze_client: MongoClient, cards_json: list[str]) -> Collection:
     cards_collection = mock_scooze_client.scooze[DbCollection.CARDS]
     for card in cards_json:
-        cards_collection.insert_one(
-            CardModelData.model_validate_json(card).model_dump(
-                mode="json",
-                by_alias=True,
-            )
-        )
+        cards_collection.insert_one(CardModelData.model_validate_json(card).model_dump(mode="json", by_alias=True))
```
Comment (JacobGinsparg) [inline tests/routers/conftest.py:22]: I guess `black` is fine with this line being long now?

**Review by JacobGinsparg — APPROVED** [review]
LGTM. Wait for @fjork3 to review since we paired on a lot of the alias debugging.

File: tests/routers/conftest.py (hunk 3 above)
Comment (iambroadband) [reply tests/routers/conftest.py:22]: I removed a trailing comma. Happy to put it back if you'd like.

File: src/scooze/models/card.py (hunk 2 above)
Comment (fjork3) [reply src/scooze/models/card.py:177]: Confirming the behavior here: when we write out, we're expecting that it'll use the first one in this list as the field name? (i.e. the JSON output will have `scryfall_id`, and that's what we want)

**Review by fjork3 — APPROVED** [review]

**Pull request merged**